R/OsModel.R

```diff
@@ -0,0 +1,171 @@
+#' Description - Log-logistic overall survival model object creator
+#'
+#' @export
+LogLogisticOs <- setClass(
+    Class = "LogLogisticOs",
+    contains = "OsModel"
+)
+
+
+#' LogLogisticModule
+#'
+#' Log logistic module helper function.
+#' @param functions
+#' @param data
+#' @param parameters
+#' @param transformed_parameters
+#' @param generated_quantities
+#' @param priors
+#' @param inits
+#' @param templated
+#' @export
+LogLogisticModule <- function(functions = "os_functions.stan",
+                              data = "os_data.stan",
+                              parameters = "os_parameters.stan",
+                              transformed_parameters = "os_transformed_parameters.stan",
+                              generated_quantities = "os_generated_quantities.stan",
+                              priors = os_prior(),
+                              inits = list(),
+                              templated = TRUE) {
+    st_mod <- StanModule(
+            functions = functions,
+            data = data,
+            parameters = parameters,
+            transformed_parameters = transformed_parameters,
+            generated_quantities = generated_quantities,
+            priors = priors,
+            inits = inits
+        )
+
+    OsModel(
+        stan = st_mod,
+        templated = templated
+    )
+}
+
+
+
+
+
+
+#' LogLogisticOs object creator
+#'
+#' Creates a log-logistic overall survival templated object
+#' @rdname LogLogisticOs-class
+#' @importFrom assertthat assert_that
+#' @export
+setMethod(
+    f = "initialize",
+    signature = "LogLogisticOs",
+    definition = function(.Object,
+                          ..., stan = LogLogisticModule(), templated) {
+        callNextMethod(
+            .Object,
+            ...,
+            stan = stan@stan,
+            templated = stan@templated
+        )
+    }
+)
+
+
+
+
+
+
+
+
+#' Parametrize TemplatedStanOs object with the selected Hazardlink
+#' @param osmod TemplatedStanOs object
+#' @param link HazardLink object
+#' @importFrom stringr str_replace
+#' @importFrom stringr str_remove_all
+#' @importFrom stringr str_replace_all
+#' @export
+setGeneric("parametrize", function(osmod, link) {
+    standardGeneric("parametrize")
+})
+
+
+
+
+
+# To be updated
+setMethod(
+    "parametrize",
+    signature(osmod = "OsModel", link = "HazardLink"),
+    function(osmod, link) {
+        newOS <- OsModel(stan = StanModule(
+            functions = osmod@stan@functions,
+            data = osmod@stan@data,
+            parameters = osmod@stan@data,
+            transformed_parameters = osmod@stan@transformed_parameters,
+            priors = osmod@stan@priors,
+            generated_quantities = osmod@stan@generated_quantities,
+            inits = osmod@stan@inits
+        ))
+
+        newOS@stan@functions <- str_replace_all(
+            string = osmod@stan@functions,
+            pattern = "<link_arguments>",
+            replacement = paste0("real ", link@parameters, ",")
```

Comment:
Should the "," be  ";" ?

Reply:
Ah apoloiges I assumed (incorrectly) these were just individual statements as opposed to function inputs. Ignore me :)